File: forum/src/main/java/com/example/forum/domain/Article.java
package com.example.forum.domain;

import jakarta.persistence.*;
import lombok.*;

import java.util.LinkedHashSet;
import java.util.Objects;
import java.util.Set;

@Getter
@ToString(callSuper = true)
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Table(indexes = {
        @Index(columnList = "title"),
        @Index(columnList = "createdAt"),
        @Index(columnList = "createdBy")
})
@Entity
public class Article extends AuditingFields {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Setter @Column(nullable = false)
    private String title;
    @Setter @Column(nullable = false, length = 10000)
    private String content;
    @Setter
    private String hashtag;

    @ToString.Exclude
    @OrderBy("createdAt DESC")
    @OneToMany(mappedBy = "article", cascade = CascadeType.ALL)
    private final Set<ArticleComment> articleComments = new LinkedHashSet<>();

    @Setter
    @ManyToOne(optional = false)
    @JoinColumn(name = "user_id")
    private UserAccount userAccount;

    @Setter
    @ManyToOne(optional = false)
    @JoinColumn(name = "user_bidder_id")
    private UserAccount userAccount2;

    private Article(UserAccount userAccount, String title, String content, String hashtag) {
        this.title = title;
        this.content = content;
        this.hashtag = hashtag;
        this.userAccount = userAccount;
    }

    public static Article of(UserAccount userAccount, String title, String content, String hashtag) {
        return new Article(userAccount, title, content, hashtag);
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Article article)) return false;
        return id != null && id.equals(article.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: migration-to-kotlin/src/main/kotlin/com/example/practice/domain/Article.kt
package com.example.practice.domain

import jakarta.persistence.*

@Table(indexes = [Index(columnList = "title"), Index(columnList = "hashtag")])
@Entity
class Article(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY) val id : Long = 0L,
    @Column(length = 10000, nullable = false) val content : String,
    @Column(length = 255, nullable = false) val title : String,
    @Column(length = 255) val hashtag : String,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_account_id")
    val userAccount : UserAccount,

    @OneToMany(mappedBy = "article", orphanRemoval = true)
    val articleComments : Set<ArticleComment> = linkedSetOf()

) : AuditingFields() {
    companion object {
        internal fun of(userAccount: UserAccount, title: String, content: String, hashtag: String) : Article {
            return Article(
                userAccount = userAccount,
                title = title,
                content = content,
                hashtag = hashtag
            )
        }
    }
}